### PR TITLE
Allow JSON to be passed as a String to SchemaMapper

### PR DESCRIPTION
**Pull request by raks81**
This is a proposed fix for #255 to allow example JSON to be passed to SchemaMapper as String. Currently it needs to be passed as an URL which is not always convenient. Fix #221 only added support for JSON schema. This PR adds support for example JSON as well.

### CODE REVIEW

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/SchemaMapper.java [hunk 1]
```diff
@@ -121,8 +121,17 @@ public JType generate(JCodeModel codeModel, String className, String packageName
         JPackage jpackage = codeModel._package(packageName);
 
         ObjectMapper mapper = new ObjectMapper();
-        JsonNode schemaNode = mapper.readTree(json);
-
+        JsonNode schemaNode = null;
+        switch (ruleFactory.getGenerationConfig().getSourceType()) {
+		case JSON:
+			JsonNode jsonNode = mapper.readTree(json);
+	        schemaNode = schemaGenerator.schemaFromExample(jsonNode);
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/SchemaMapper.java]: Indentation problem here

File: jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/SchemaMapper.java [hunk 2]
```diff
@@ -121,8 +121,17 @@ public JType generate(JCodeModel codeModel, String className, String packageName
         JPackage jpackage = codeModel._package(packageName);
 
         ObjectMapper mapper = new ObjectMapper();
-        JsonNode schemaNode = mapper.readTree(json);
-
+        JsonNode schemaNode = null;
+        switch (ruleFactory.getGenerationConfig().getSourceType()) {
```
Comment (joelittlejohn) [inline jsonschema2pojo-core/src/main/java/org/jsonschema2pojo/SchemaMapper.java]: For this simple case, I think `if`/`else` would be neater.

**Comment by joelittlejohn** [comment]
Thanks a lot for submitting this, I've made some minor review comments. If you fix these up, I will merge and release this in 0.4.19.

**Comment by raks81** [comment]
@joelittlejohn I have fixed code as per your comments. Please review. Thanks!

**Comment by joelittlejohn** [comment]
Thanks @raks81. One final step, please squash these commits into one.

**Comment by raks81** [comment]
@joelittlejohn I have squashed the commmits. Please review. Thanks!

**Comment by joelittlejohn** [comment]
:+1:

**Pull request merged**